applications/GeoMechanicsApplication/custom_processes/find_neighbour_elements_of_conditions_process.cpp

```diff
@@ -214,40 +176,103 @@ void FindNeighbourElementsOfConditionsProcess::Execute()
+    
     //check that all of the conditions belong to at least an element. Throw an error otherwise (this is particularly useful in mpi)
-    AllVisited = true;
+    all_conditions_visited = true;
     for (auto& rCond : mrModelPart.Conditions()) {
         if (rCond.IsNot(VISITED)) {
-            AllVisited = false;
+            all_conditions_visited = false;
             KRATOS_INFO("Condition without any corresponding element, ID ") << rCond.Id() << std::endl;
         }
     }
 
-    KRATOS_ERROR_IF_NOT(AllVisited) << "Some conditions found without any corresponding element" << std::endl;
+    KRATOS_ERROR_IF_NOT(all_conditions_visited) << "Some conditions found without any corresponding element" << std::endl;
 
     KRATOS_CATCH("")
 }
 
+bool FindNeighbourElementsOfConditionsProcess::CheckIfAllConditionsAreVisited() const
+{
+    for (auto& r_cond : mrModelPart.Conditions()) {
+        if (r_cond.IsNot(VISITED)) {
+            return false;
+        }
+    }
+    return true;
+}
+
+void FindNeighbourElementsOfConditionsProcess::CheckIf1DElementIsNeighbour(hashmap& rFacesMap)
+{
+    // Now loop over all elements and check if one of the faces is in the "FacesMap"
+    for (auto itElem = mrModelPart.ElementsBegin(); itElem != mrModelPart.ElementsEnd(); ++itElem) {
+        const auto& r_geometry_element = itElem->GetGeometry();
+
+	    // for 1D elements, the edge geometry is the same as the element geometry 
+	    if (r_geometry_element.LocalSpaceDimension() == 1)
+	    {
+            const auto rBoundaryGeometries = PointerVector(r_geometry_element.GenerateEdges());
+
+            for (IndexType iFace = 0; iFace < rBoundaryGeometries.size(); ++iFace) {
+                DenseVector<int> FaceIds(rBoundaryGeometries[iFace].size());
+
+                // get edge ids
+                for (IndexType iNode = 0; iNode < FaceIds.size(); ++iNode) {
+                    FaceIds[iNode] = rBoundaryGeometries[iFace][iNode].Id();
+                }
+
+                hashmap::iterator itFace = rFacesMap.find(FaceIds);
+
+                if (itFace != rFacesMap.end()) {
+                    // condition is found!
+                    // but check if there are more than one condition on the element
+                    CheckForMultipleConditionsOnElement(rFacesMap, itFace, itElem);
+
+                }
+            }
+        }
+    }
+}
+
+
+void FindNeighbourElementsOfConditionsProcess::CheckForMultipleConditionsOnElement(hashmap& rFacesMap, hashmap::iterator& rItFace, 
+    PointerVector<Element>::iterator pItElem)
+{
+
+	const std::pair<hashmap::iterator, hashmap::iterator> face_pair = rFacesMap.equal_range(rItFace->first);
+    for (hashmap::iterator it = face_pair.first; it != face_pair.second; ++it) {
+        std::vector<Condition::Pointer>& r_conditions = it->second;
+
+        GlobalPointersVector< Element > vector_of_neighbours;
+        vector_of_neighbours.resize(1);
+        vector_of_neighbours(0) = Element::WeakPointer(*pItElem.base());
```

Comment:
```suggestion
	const std::pair<hashmap::iterator, hashmap::iterator> face_pair = rFacesMap.equal_range(rItFace->first);
        for (hashmap::iterator it = face_pair.first; it != face_pair.second; ++it) {
            std::vector<Condition::Pointer>& r_conditions = it->second;

            GlobalPointersVector< Element > vector_of_neighbours;
            vector_of_neighbours.resize(1);
            vector_of_neighbours(0) = Element::WeakPointer(*pItElem.base());
```
Corrected the alignment